REFACTORING_RECOMMENDATIONS.md

```diff
@@ -0,0 +1,1205 @@
+    }
+
+    syncCurrentPhaseWithPhaseList(): void {
+        if (this.state.phaseList === null || this.state.phaseList.length === 0) {
+            this.state.currentPhase = null;
+        }
+    }
+
+    assertCurrentPhase(currentPhase: number | null): void {
+        if (this.state.phaseList === null || this.state.phaseList.length === 0) {
+            if (currentPhase === null) {
+                return;
+            }
+            throw new ImplementationError(
+                "Cannot set current phase to an other value than null when phase list is empty",
+            );
+        }
+
+        if (currentPhase === null || currentPhase < 0 || currentPhase >= this.state.phaseList.length) {
+            throw new ImplementationError(
+                `Current phase ${currentPhase} is out of bounds for phase list of length ${this.state.phaseList.length}`,
+            );
+        }
+    }
+
+    assertOperationalState(newState: number, oldState: number): void {
+        if (!this.state.operationalStateList.some(({ operationalStateId }) => operationalStateId === newState)) {
+            throw new ImplementationError(
+                `Cannot set operational state id ${newState} as it is not in the operational state list`,
+            );
+        }
+        if (
+            oldState === OperationalState.OperationalStateEnum.Error &&
+            this.state.operationalError.errorStateId !== OperationalState.ErrorState.NoError &&
+            newState !== OperationalState.OperationalStateEnum.Error
+        ) {
+            this.state.operationalError = { errorStateId: OperationalState.ErrorState.NoError };
+        }
+    }
+
+    handleOperationalError(operationalError: OperationalState.ErrorStateStruct): void {
+        if (operationalError.errorStateId === OperationalState.ErrorState.NoError) {
+            return;
+        }
+
+        if (this.state.operationalState !== OperationalState.OperationalStateEnum.Error) {
+            this.state.operationalState = OperationalState.OperationalStateEnum.Error;
+        }
+
+        this.events.operationalError.emit({ errorState: operationalError }, this.context);
+    }
+}
+```
+
+**Refactor Servers**:
+
+```typescript
+// Before: /packages/node/src/behaviors/operational-state/OperationalStateServer.ts
+export class OperationalStateServer extends OperationalStateBehavior {
+    #assertOperationalStateList() { /* 10 lines */ }
+    #syncCurrentPhaseWithPhaseList() { /* 5 lines */ }
+    #assertCurrentPhase(currentPhase: number | null) { /* 15 lines */ }
+    #assertOperationalState(newState: number, oldState: number) { /* 15 lines */ }
+    #handleOperationalError(operationalError: OperationalState.ErrorStateStruct) { /* 11 lines */ }
+}
+
+// After:
+import { OperationalStateValidation } from "../utils/OperationalStateValidation.js";
+
+export class OperationalStateServer extends OperationalStateBehavior {
+    #validation = new OperationalStateValidation(this.state, this.events, this.context);
```

Comment:
This will create even if unused. Instead create on demand as needed?

Also, don't see benefit of passing ketosis as the separate parameters. Just pass `this` instead